managed/src/main/java/com/yugabyte/yw/controllers/BackupsController.java

```diff
@@ -163,37 +165,47 @@ public Result restore(UUID customerUUID, UUID universeUUID) {
     return ApiResponse.success(resultNode);
   }
 
-  public Result delete(UUID customerUUID, UUID backupUUID) {
+  public Result delete(UUID customerUUID) {
+    List<UUID> validBackups = new ArrayList<UUID>();
+    List<String> taskUUIDList = new ArrayList<String>();
+    ObjectNode formData = (ObjectNode) request().body().asJson();
     Customer customer = Customer.get(customerUUID);
     if (customer == null) {
       String errMsg = "Invalid Customer UUID: " + customerUUID;
       return ApiResponse.error(BAD_REQUEST, errMsg);
     }
-    Backup backup = Backup.get(customerUUID, backupUUID);
-    if (backup == null) {
-      String errMsg = "Invalid Backup UUID: " + customerUUID;
-      return ApiResponse.error(BAD_REQUEST, errMsg);
-    }
-    if (backup.state != Backup.BackupState.Completed) {
-      String errMsg = String.format("Cannot delete backup %s since it hasn't been completed",
-                                    backupUUID.toString());
-      return ApiResponse.error(BAD_REQUEST, errMsg);
+    for (JsonNode backupUUID : formData.get("backupUUID")) {
+      Backup backup = Backup.get(customerUUID, UUID.fromString(backupUUID.asText()));
+      if (backup == null) {
+          LOG.info("Can not delete {} backup as it is not present in the database.",
+              backupUUID.asText());
+      }
+      else {
+        validBackups.add( UUID.fromString(backupUUID.asText()));
+      }
     }
-    DeleteBackup.Params taskParams = new DeleteBackup.Params();
-    taskParams.customerUUID = customerUUID;
-    taskParams.backupUUID = backupUUID;
-    UUID taskUUID = commissioner.submit(TaskType.DeleteBackup, taskParams);
-    LOG.info("Saved task uuid {} in customer tasks for backup {}.",
-             taskUUID, backupUUID);
-    CustomerTask.create(customer,
-        backupUUID,
-        taskUUID,
-        CustomerTask.TargetType.Backup,
-        CustomerTask.TaskType.Delete,
-        "Backup");
     ObjectNode resultNode = Json.newObject();
-    resultNode.put("taskUUID", taskUUID.toString());
-    Audit.createAuditEntry(ctx(), request(), taskUUID);
+    for (UUID uuid : validBackups) {
+      Backup backup = Backup.get(customerUUID, uuid);
+      
+      if (backup.state != Backup.BackupState.Completed) {
+        LOG.info("Can not delete {} backup as it is already in progress", uuid);
```

Comment:
Not `already` but `still`.

Reply:
Also, as I've already said about two loops instead on the initial one, this check should be moved to the first loop.